### PR TITLE
feat: Use wingetcreate for releasing to WinGet

### LINKED ISSUE #497: Use wingetcreate for releasing new versions to WinGet
The current workflow file uses an existing manifest file and simply replaces certain fields when creating the new manifest. While this works, it leaves the WinGet manifests in non-standard state.

1) Manifests created do not have the `# yaml-language-server: $schema=<url>` header
2) Tags are not the same format as all other manifests
3) Manifest schema is not updated to the latest

It would be preferred to use `wingetcreate.exe` in your releaser action so that the manifest content is formatted consistently and the latest manifest version is used.

### PR DESCRIPTION
**Pull request by Trenly**
* Resolves #497

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -383,108 +383,10 @@ jobs:
...
-          cd winget-pkgs
-          gh pr create --repo microsoft/winget-pkgs \
-            --base master \
-            --head dotenvx:dotenvx-dotenvx-${{ env.CLEAN_VERSION }} \
-            --title "New version: dotenvx.dotenvx ${{ env.CLEAN_VERSION }}" \
-            --body "New version: dotenvx.dotenvx ${{ env.CLEAN_VERSION }}"
-
+      - name: Submit using wingetcreate
+        run: | 
+          iwr https://aka.ms/wingetcreate/latest -OutFile wingetcreate.exe
```
Comment (motdotla) [inline .github/workflows/ci.yml:388]: thanks for this PR! looks potentially very helpful (removing a lot of boilerplate)

but can you point me to docs on this url? i'm hesitant to trust it initially as it downloads an executable remotely.

Comment (Trenly) [reply .github/workflows/ci.yml:388]: It's what PowerToys uses in their release - https://github.com/microsoft/PowerToys/blob/main/.github/workflows/package-submissions.yml

Comment (Trenly) [reply .github/workflows/ci.yml:388]: @denelon can potentially point to docs directly

Comment (motdotla) [reply .github/workflows/ci.yml:388]: that's pretty trustworthy :). thanks. i'll try and test this out shortly.

Comment (denelon) [reply .github/workflows/ci.yml:388]: Let me know if you have any concerns :) I'm happy to help.

Comment (denelon) [reply .github/workflows/ci.yml:388]: We're hoping to get WinGet in a good state on the Windows images so you get the benefit of the package manager with hash validation and all the other goodies. Initially, it's likely to have an outdated version that won't work without some fiddling.

Comment (motdotla) [reply .github/workflows/ci.yml:388]: ok, i think it worked: https://github.com/dotenvx/dotenvx/actions/runs/12439823801/job/34734350769

where do i go to view it now that it is not a PR request?

Comment (motdotla) [reply .github/workflows/ci.yml:388]: nevermind. i see it here: https://github.com/microsoft/winget-pkgs/pull/200439

thanks for this guys! anything else you need from me let me know.

Comment (Trenly) [reply .github/workflows/ci.yml:388]: Looks great!

**Comment by motdotla** [comment]
@Trenly was there a recent change to the `wingetcreate` tool? 

I didn't make any CI changes but failing here on our latest release 1.33.0: https://github.com/dotenvx/dotenvx/actions/runs/12753460407/job/35546116519

Rest of my automated deployments go out smoothly - exe, binaries, etc.

stacktrace:

```
Telemetry Settings
------------------
The Windows Package Manager Manifest Creator collects usage data in order to improve your experience.
The data is anonymous and collected only by Microsoft.
By default, telemetry is enabled but can be disabled by running `wingetcreate settings` and editing your settings file.

Retrieving latest manifest for dotenvx.dotenvx
Downloading and parsing: https://github.com/dotenvx/dotenvx/releases/download/v1.33.0/dotenvx-1.33.0-windows-x86_64.zip...
GitHub URL detected. The CLI has automatically filled some manifest fields.
Generating a preview of your manifests...
Version manifest preview:
# Created using wingetcreate 1.9.3.0
# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.9.0.schema.json

PackageIdentifier: dotenvx.dotenvx
PackageVersion: 1.33.0
DefaultLocale: en-US
ManifestType: version
ManifestVersion: 1.9.0

Installer manifest preview:
# Created using wingetcreate 1.9.3.0
# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.9.0.schema.json

PackageIdentifier: dotenvx.dotenvx
PackageVersion: 1.33.0
InstallerLocale: en-US
InstallerType: zip
NestedInstallerType: portable
NestedInstallerFiles:
- RelativeFilePath: dotenvx.exe
  PortableCommandAlias: dotenvx
UpgradeBehavior: uninstallPrevious
Installers:
- Architecture: x64
  InstallerUrl: https://github.com/dotenvx/dotenvx/releases/download/v1.33.0/dotenvx-1.33.0-windows-x86_64.zip
  InstallerSha256: 98DD8B8C8464B996002A5D03E6AC1A1979AD30D1EE0631A9D4E847C331C68DED
ManifestType: installer
ManifestVersion: 1.9.0
ReleaseDate: 2025-01-13

Default locale manifest preview:
# Created using wingetcreate 1.9.3.0
# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.9.0.schema.json

PackageIdentifier: dotenvx.dotenvx
PackageVersion: 1.33.0
PackageLocale: en-US
Publisher: dotenvx
PublisherUrl: https://dotenvx.com/
PublisherSupportUrl: https://github.com/dotenvx/dotenvx/issues
Author: motdotla
PackageName: dotenvx
PackageUrl: https://github.com/dotenvx/dotenvx
License: BSD-3-Clause
ShortDescription: a better dotenv–from the creator of dotenv
Description: a better dotenv–from the creator of dotenv
Moniker: dotenvx
Tags:
- env
- dotenv
- environment
- variables
- dotenvx
- cli
- encryption
ReleaseNotesUrl: https://github.com/dotenvx/dotenvx/releases/tag/v1.33.0
ManifestType: defaultLocale
ManifestVersion: 1.9.0


Manifest saved to D:\a\dotenvx\dotenvx\manifests\d\dotenvx\dotenvx\1.33.0

Manifest validation succeeded: True

Submitting pull request for manifest...

System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.WingetCreateCLI.Commands.BaseCommand.GitHubSubmitManifests(Manifests manifests, String prTitle, Boolean shouldReplace, String replaceVersion)
   at Microsoft.WingetCreateCLI.Commands.UpdateCommand.ExecuteManifestUpdate(List`1 latestManifestContent, CommandExecutedEvent commandEvent)
   at Microsoft.WingetCreateCLI.Commands.UpdateCommand.Execute()
   at Microsoft.WingetCreateCLI.Program.Main(String[] args)
```

Screenshot 2025-01-13 at 10 41 45 AM

**Comment by mdanish-kh** [comment]
@motdotla As an immediate workaround - you can manually sync your winget-pkgs fork and then kick off the run again

**Comment by mdanish-kh** [comment]
The bug should be resolved once we have a new winget-create release

**Comment by motdotla** [comment]
> The bug should be resolved once we have a new winget-create release

cool, i'll just wait for that and then release. no rush. thanks guys!

**Comment by denelon** [comment]
Sorry about the problem. It looks like it's potentially a breaking change in octokit, but I'm not sure. Either way we'll get a release out in the next couple of days with the fix 😊

**Comment by mdanish-kh** [comment]
@motdotla - There's been a new release of WingetCreate ([v1.9.4.0](https://github.com/microsoft/winget-create/releases/tag/v1.9.4.0)) which should hopefully address the issue. Let us know if you still bump into it

**Comment by motdotla** [comment]
@mdanish-kh it worked! thanks! 

https://github.com/dotenvx/dotenvx/actions/runs/12753460407/job/35626921860

**Pull request merged**